src/prop-validators/arrayWithLength.js

```diff
@@ -0,0 +1,57 @@
+import propTypes from 'prop-types'
+
+const arrayWithLengthFactory = ({
+    min = 0,
+    max = Infinity,
+    propType,
+    isRequired,
+}) => (props, propName, componentName) => {
+    const arr = props[propName]
+
+    if (isRequired && typeof arr === 'undefined') {
+        return new Error(`${propName} is required.`)
+    }
+
+    if (arr && !Array.isArray(arr)) {
+        return new Error(`${propName} is not an array.`)
+    }
+
+    if (arr && arr.length > max) {
+        return new Error(
+            `${propName} array exceeds the maximum length of ${max}`
```

Comment:
Might be useful to know the actual length of the supplied array for debugging purposes